src/mesh/domain.hpp

```diff
@@ -45,8 +45,20 @@ struct IndexRange {
 //  - - - - - - - - - -   |
 //  |  |           |  |   |
 //  - - - - - - - - - -   v
-//
-enum class IndexDomain { entire, interior };
+
+// inner/outer_x* are ranges for boundaries
+// and cover the ghost zones on each side of the
+// meshblock
+enum class IndexDomain {
+  entire,
+  interior,
+  inner_x1,
+  outer_x1,
+  inner_x2,
+  outer_x2,
+  inner_x3,
+  outer_x3
```

Comment:
I think, `inner` and `outer` might be a bit misleading, inner - makes me think it is a domain within the interior while outer makes me think it is located on the outer edge. I think bottom and top would work better, or front/back, anterior/posterior, fore/aft, any of these combinations would be a bit more consistent with the domain enum. If you had a separate enum for edge then I can see inner and outer working.

Reply:
The reason I went with `inner` and `outer` is that 's the naming convention elsewhere throughout the code. I.e., in the `parthinput` files, that's what those are called. I'm open to changing it, but I would rather that be a code-wide change that happens in a later PR after some discussion. Would that be ok?

Reply:
https://github.com/lanl/parthenon/issues/354

Reply:
It would be nice if there were a figure to help illustrate these different zones.

Reply:
Yeah I had one in there at some point, but I think it was difficult to parse. I'll try my hand at it again.

Reply:
Figure added.